app/validators/blocks/unrelated_block_validator.py

```diff
@@ -0,0 +1,23 @@
+from app.validators.blocks.block_validator import BlockValidator
+
+
+class UnrelatedBlockValidator(BlockValidator):
+    ACTION_PARAMS_MISSING = "RemoveUnrelatedRelationships and AddUnrelatedRelationships actions must be present"
+
+    def validate(self):
+        super().validate()
+        self.validate_answer_actions()
+        return self.errors
+
+    def validate_answer_actions(self):
+        expected_actions = ["AddUnrelatedRelationships", "RemoveUnrelatedRelationships"]
+        questions = self.questionnaire_schema.get_all_questions_for_block(self.block)
+        for question in questions:
+            actions = []
+            for answer in question["answers"]:
+                for option in answer["options"]:
+                    if "action" in option:
+                        actions.append(option["action"]["type"])
```

Comment:
Ideal for a comprehension?
```suggestion
            actions = [
                option["action"]["type"]
                for answer in question["answers"]
                for option in answer["options"]
                if "action" in option
            ]
```

Reply:
👍